FirebaseFunctions/Sources/Internal/FunctionsSerializer.swift

```diff
@@ -14,17 +14,10 @@
 
 import Foundation
 
-private enum Constants {
-  static let longType = "type.googleapis.com/google.protobuf.Int64Value"
-  static let unsignedLongType = "type.googleapis.com/google.protobuf.UInt64Value"
-  static let dateType = "type.googleapis.com/google.protobuf.Timestamp"
-}
-
 extension FunctionsSerializer {
   enum Error: Swift.Error {
     case unsupportedType(typeName: String)
-    case unknownNumberType(charValue: String, number: NSNumber)
-    case invalidValueForType(value: String, requestedType: String)
+    case failedToParseWrappedNumber(WrappedNumber)
```

Comment:
`WrappedNumber` can be made `private` if this error cases uses plain strings. I decided to go with the actual object to avoid the extra transformations, but I still have some reservations.

Reply:
I think this could possibly be a subtle breaking change since the number is currently bubbling up into the public API surface as an `NSNumber` and now it's an `any Numeric `. 

<img width="1012" alt="Screenshot 2025-05-30 at 4 42 03 PM" src="https://github.com/user-attachments/assets/3a10a129-7ae4-4e42-a967-046669c70ab8" />

<img width="880" alt="Screenshot 2025-05-30 at 4 43 05 PM" src="https://github.com/user-attachments/assets/7ec82786-708d-44f6-b84c-36116cd889e3" />

Can an `any Numeric` cast to `NSNumber`?

Reply:
Based on this (https://forums.swift.org/t/with-only-t-numeric-type-how-to-create-an-nsnumber/47663/3) I believe the answer is yes except for `Bool` and `CChar`

Reply:
And since 
```swift
private func unwrapNumber(_ wrapped: WrappedNumber) throws(Error) -> any Numeric ```

only returns an Int or UInt, which both bridge to NSNumber, it should be OK